### PR TITLE
Converting check for understanding

### PR DESCRIPTION
**Pull request by rudri182**
Created some questions in the check for understanding part of the stack.md outline. The issue is #700

### CODE REVIEW

**Comment by praveenscience** [comment]
@rudri182 This PR has a lot of conflicts. Are you working off an old version? Can you try rebasing your branch and resolving conflicts?

File: agile-development/scrum.md [hunk 1]
```diff
@@ -5,12 +5,12 @@
 Example: 60 - 120 minutes
 
 ### Prerequisites
-HTML  
-CSS  
-Git  
-GitHub  
-Editor  
-Desire to learn  
+- [HTML](https://github.com/Techtonica/curriculum/blob/master/html/html.md)
```
Comment (praveenscience) [inline agile-development/scrum.md]: Please use relative paths. Example, replace:
`https://github.com/Techtonica/curriculum/blob/master/html/html.md`
With:
`../html/html.md`

**Comment by rudri182** [comment]
Removed the unwanted files that were pushed by mistake. 😄 Now I will resolve conflicts

**Comment by praveenscience** [comment]
Ha ha, thanks @rudri182.

**Comment by rudri182** [comment]
Do I need to resolve it now? @praveenscience

File: data-structures/stack.md [hunk 2]
```diff
@@ -34,17 +33,20 @@ Make sure to mention these things:
...
 
 ### Common Mistakes / Misconceptions
-- Array and stack seem similar at first glance. While stack can be implemented using array, the data in array can be access randomly whereas stack must be access according to order.
+
+- Array and stack seem similar at first glance. While stack can be implemented using array, the data in array can be accessed randomly, whereas stack must be accessed according to order.
 
 ### Guided Practice
+
 Explain and discuss as a class the steps involved in writing a stack structure, including:
 - Constructor
```
Comment (praveenscience) [inline data-structures/stack.md]: Give a new line before starting a list.

File: data-structures/stack.md [hunk 3]
```diff
@@ -60,4 +62,23 @@ var Stack = function() {}
...
 Find a partner and show each other your own Stack class. Explain how the class you wrote works.
+
+Make a pair of two and ask the following questions to each-other.
+
+- What is difference between stack and simple array?
+- Which method does the stack follows?(LIFO)
+- Which methods are used in the stack data structure?
+- What is run time complexity of the stack?
+- How to make simple class as a stack class?
+- What is the meaning of '_'(*underscore*) in the begining of the variable name? Refer [Playing with data structures in Javascript — Stack](https://blog.cloudboost.io/playing-with-data-structures-in-javascript-stack-a55ebe50f29d)
```
Comment (praveenscience) [inline data-structures/stack.md]: Space after underscore? (`'_'`)

**Comment by rudri182** [comment]
How to resolve conflicts? There is no link here. 
image
Or these conflicts are solved?

**Comment by praveenscience** [comment]
@rudri182 I just pulled your branch and checked the source. The problem with the current branch you are working on is completely out of sync and GitHub / Git engine isn't able to properly merge the changes, so you might need to cherry pick the changes and push them. I can help you with it if you find cherry-picking and applying the changes difficult. But yea, at this point, the merging can only be done through command line not using any GUI.

File: deploying/deploying.md [hunk 4]
```diff
@@ -4,15 +4,15 @@
...
 
-Node.js and npm must be installed
-An existing Node.js app
-JS I - VI
-Node
-Express
-MongoDB
+- Node.js and npm must be installed
+- An existing Node.js app
+- [JS I - VI](https://github.com/Techtonica/curriculum/tree/master/javascript)
```
Comment (praveenscience) [inline deploying/deploying.md]: Please use relative paths. Example, replace:
`https://github.com/Techtonica/curriculum/blob/master/html/html.md`
With:
`../html/html.md`

File: express-js/express.md [hunk 5]
```diff
@@ -4,8 +4,8 @@
 4 hours, including lab exercise
 
 ### Prerequisites
-- How the Internet Works
-- Node.js
+- [How the Internet Works](https://github.com/Techtonica/curriculum/blob/master/requests-and-responses/requests-and-responses.md)
```
Comment (praveenscience) [inline express-js/express.md]: Please use relative paths. Example, replace:
`https://github.com/Techtonica/curriculum/blob/master/html/html.md`
With:
`../html/html.md`

File: functional-programming/FP.md [hunk 6]
```diff
@@ -5,18 +5,15 @@
...
-
-- independent practice: 30 min 
+- independent practice: 30 min
 
 ### Prerequisites
-- JavaScript basics
-- `for` loops
-- arrays
-- Object-oriented programming
+- [JavaScript basics](https://github.com/Techtonica/curriculum/tree/master/javascript)
```
Comment (praveenscience) [inline functional-programming/FP.md]: Please use relative paths. Example, replace:
`https://github.com/Techtonica/curriculum/blob/master/html/html.md`
With:
`../html/html.md`

File: node-js/node-js.md [hunk 7]
```diff
@@ -4,11 +4,11 @@
 90 minutes
 
 ### Prerequisites
-- JavaScript I - VI lessons
-- How the Internet Works
-- AJAX
-- Git and Version Control
-- jQuery
+- [JavaScript I - VI lessons](https://github.com/Techtonica/curriculum/tree/master/javascript)
```
Comment (praveenscience) [inline node-js/node-js.md]: Please use relative paths. Example, replace:
`https://github.com/Techtonica/curriculum/blob/master/html/html.md`
With:
`../html/html.md`

File: open-source/open-source.md [hunk 8]
```diff
@@ -4,8 +4,9 @@
 1 hour
 
 ### Prerequisites
-- "Git and version control" lesson
-- "Using GitHub for Code Storage" lesson
+- ["Git and version control" lesson](https://github.com/Techtonica/curriculum/blob/master/git-version-control/git-version-control.md)
```
Comment (praveenscience) [inline open-source/open-source.md]: Please use relative paths. Example, replace:
`https://github.com/Techtonica/curriculum/blob/master/html/html.md`
With:
`../html/html.md`

File: projects/eventonica-project.md [hunk 9]
```diff
@@ -2,20 +2,20 @@
 # Eventonica Project
 
 ### Languages
-- JavaScript
-- SQL query
+- [SQL query](https://github.com/Techtonica/curriculum/blob/master/databases/sql.md)
```
Comment (praveenscience) [inline projects/eventonica-project.md]: Please use relative paths. Example, replace:
`https://github.com/Techtonica/curriculum/blob/master/html/html.md`
With:
`../html/html.md`

File: projects/minimal-api-project.md [hunk 10]
```diff
@@ -1,10 +1,10 @@
 ## Week 7 Assessment - Build a Minimal API Project
 
 ### Languages
-- JS
-- Git
-- bash
-- SQL
+- [JS](https://github.com/Techtonica/curriculum/tree/master/javascript)
```
Comment (praveenscience) [inline projects/minimal-api-project.md]: Please use relative paths. Example, replace:
`https://github.com/Techtonica/curriculum/blob/master/html/html.md`
With:
`../html/html.md`

File: react-js/react.md [hunk 11]
```diff
@@ -4,8 +4,8 @@
 8 Hours
 
 ### Prerequisites
-* Javascript I - V
-* jQuery Topic Outline
+* [Javascript I - VI](https://github.com/Techtonica/curriculum/tree/master/javascript)
```
Comment (praveenscience) [inline react-js/react.md]: Please use relative paths. Example, replace:
`https://github.com/Techtonica/curriculum/blob/master/html/html.md`
With:
`../html/html.md`

File: recursion/recursion.md [hunk 12]
```diff
@@ -7,11 +7,10 @@ About 4.5 hours
 - 240 minutes for Independent Practice & Challenge
 
 ### Prerequisites
-- JS 1 lesson
-- JS 2 lesson
-- JS 3 lesson
-- JS 4 lesson
-- Score of __15__ or higher at Coding Challege Assesment, Level 1
+- (JavaScript 1 lesson)[https://github.com/Techtonica/curriculum/blob/master/javascript/javascript-1.md]
```
Comment (praveenscience) [inline recursion/recursion.md]: Please use relative paths. Example, replace:
`https://github.com/Techtonica/curriculum/blob/master/html/html.md`
With:
`../html/html.md`

File: refactoring/refactoring.md [hunk 13]
```diff
@@ -7,21 +7,21 @@
    - 45 mins Independent Practice & Group Practice
 
 ### Prerequisites
-Basic understanding of Javascript.
-"Writing Readable Code" lesson
-"Debugging" lesson
+- Basic understanding of [JavaScript](https://github.com/Techtonica/curriculum/tree/master/javascript).
```
Comment (praveenscience) [inline refactoring/refactoring.md]: Please use relative paths. Example, replace:
`https://github.com/Techtonica/curriculum/blob/master/html/html.md`
With:
`../html/html.md`

File: refactoring/refactoring.md [hunk 14]
```diff
@@ -7,21 +7,21 @@
...
+- [Writing Readable Code lesson](https://github.com/Techtonica/curriculum/blob/master/writing-readable-code/writing-readable-code.md)
+- [Debugging lesson](https://github.com/Techtonica/curriculum/blob/master/debugging/debugging.md)
 
 ### Motivation
 To encourage quality code practices and being able to identify improvements that can be made to existing code.
 
 ### Objectives
-**Participants will be able to:**: 
+**Participants will be able to:**:
```
Comment (praveenscience) [inline refactoring/refactoring.md]: Double `:`

**Comment by rudri182** [comment]
I googled it but it is a little bit difficult for me to understand `cherry-pick` as a beginner. If you can help me and direct me to what should I do then I will be happy to do it. 😊

File: roles-in-tech/machine-learning.md [hunk 15]
```diff
@@ -11,6 +11,7 @@ Example: 30-45 minutes
 ### Prerequisites
 
 Here are links to lessons that should be completed before this lesson:
+- [Data Science](https://github.com/Techtonica/curriculum/blob/master/roles-in-tech/data-science.md)
```
Comment (praveenscience) [inline roles-in-tech/machine-learning.md]: Please use relative paths. Example, replace:
`https://github.com/Techtonica/curriculum/blob/master/html/html.md`
With:
`../html/html.md`

File: sass/sass.md [hunk 16]
```diff
@@ -9,8 +9,8 @@
 - 4 Hours Total ~
 
 ### Prerequisites
-- HTML
-- CSS
+- [HTML](https://github.com/Techtonica/curriculum/blob/master/html/html.md)
```
Comment (praveenscience) [inline sass/sass.md]: Please use relative paths. Example, replace:
`https://github.com/Techtonica/curriculum/blob/master/html/html.md`
With:
`../html/html.md`

**Comment by praveenscience** [comment]
@rudri182 Sorry ya, just finished reviewing, and found many small quick fixes. Would you like to check them out too?

**Comment by rudri182** [comment]
> @rudri182 Sorry ya, just finished reviewing, and found many small quick fixes. Would you like to check them out too?

Yes, definitely I would like to change it.

File: data-structures/stack.md (hunk 3 above)
Comment (rudri182) [reply data-structures/stack.md]: May I make space after underscore? I didn't understand what is the change.

Comment (praveenscience) [reply data-structures/stack.md]: Very minor one. Give a space before `(`.

File: express-js/express.md (hunk 5 above)
Comment (rudri182) [reply express-js/express.md]: If I change the path like `..requests-and-responses/requests-and-responses.md` than it is showing _page not found error_. I am confused

Comment (praveenscience) [reply express-js/express.md]: Lemme quickly check...

File: recursion/recursion.md (hunk 12 above)
Comment (praveenscience) [reply recursion/recursion.md]: This is wrong syntax too! 😔 Swap the `(text)[https://path/]` to `[text](https://path/)`, use an editor like Typora.io to get these sorted. 😇

**Comment by rudri182** [comment]
Is the file in which I am changing the  CFU section old? Or it is up-to-date with a master?

**Comment by praveenscience** [comment]
@rudri182  I kinda edited in your branch to show how it works. Look at my commit and feel free to hard reset it.
This is the link: https://github.com/rudri182/curriculum/blob/converting-check-for-uinderstanding/refactoring/refactoring.md
This is the change: https://github.com/rudri182/curriculum/commit/1ce4efc833b1091b0c044394494901ae43c95448

File: refactoring/refactoring.md [hunk 17]
```diff
@@ -7,21 +7,21 @@
    - 45 mins Independent Practice & Group Practice
 
 ### Prerequisites
-Basic understanding of Javascript.
-"Writing Readable Code" lesson
-"Debugging" lesson
+- Basic understanding of [JavaScript](https://github.com/Techtonica/curriculum/tree/master/javascript).
+- [Writing Readable Code lesson](../writing-readable-code/writing-readable-code.md)
```
Comment (praveenscience) [inline refactoring/refactoring.md]: @rudri182 For an example, I have done this for you. Please check this.

File: express-js/express.md (hunk 5 above)
Comment (praveenscience) [reply express-js/express.md]: @rudri182 It's not `..reques...` it's `../reques...`. The `../` is going to a parent directory or path... 😇

Comment (rudri182) [reply express-js/express.md]: 👍

File: agile-development/scrum.md (hunk 1 above)
Comment (rudri182) [reply agile-development/scrum.md]: Do I need to change every path to a relative path?

File: roles-in-tech/machine-learning.md (hunk 15 above)
Comment (rudri182) [reply roles-in-tech/machine-learning.md]: This file is not on this branch so I am unable to edit it. Maybe it is removed by mistake. If it is necessary to do changes I put it here by copy-pasting from master.

**Comment by rudri182** [comment]
I have made some changes in the files and pushed it here. Please review it and suggest if there are any changes. ✏️

**Comment by alodahl** [comment]
@rudri182  git cherry-pick is actually pretty easy.  If I were you, I would:
1. `git commit` any changes you may have made since the last commit, and write down the first 7 numbers of the commit sha.
1. `git checkout master`
1. `git pull`
1. `git checkout -b check-for-understanding`
1.  git cherry-pick 1d6383b
1. git cherry-pick <second-commit-sha-you-wrote-down>

I haven't tested it, but I think that should work.  If you don't have any problems, push up your new branch, make a PR with it and see if it's what you want to merge.

**Comment by rudri182** [comment]
image

I am not able to cherry-pick the changes. I tried so many times to resolve conflicts but the same problem occurs.

image

**Comment by rudri182** [comment]
If I try to commit with git `cherry-pick --abort` and then cherry pick it then also it is showing the same error.
So I am very much confused and the same thing is appearing. If this is messed up then may I delete these both branches and make a new one?

**Comment by saberprashant** [comment]
Hi @rudri182, if you want to delete both branches and make a new one, then you have to paste all the changes one by one to the new branch. What @alodahl and @praveenscience suggesting is, make a new branch from master(after taking a pull):
`git checkout -b <new-branch>`

And then, start cherry picking the commits from the oldest ones (From your 1st commit to the latest ones):
`git cherry-pick <sha-of-commit>`

And, if this is causing you the same problem, then it might be possible that one of your cherry-pick commits has the changes which are actually causing conflicts (as you said, you added some wrong files), so please try to add those changes manually.

Thanks!

**Comment by alodahl** [comment]
Rudri, have you been trying to simply commit stacks.md?

**Comment by rudri182** [comment]
> Rudri, have you been trying to simply commit stacks.md?

I have cherry-picked the commit for it but I am not sure because every time I resolved the conflicts the same conflict message occurred as I have posted before. I then pushed the files and resolved the conflicts from the new PR. So maybe it can be simply committed but it was showing in `CHERRY-PICKING` mode in the bash.

**Comment by alodahl** [comment]
If it's just one file, just copy it and start a new branch from the current master!

**Comment by alodahl** [comment]
Your check for understandings and your prerequisite sections should be in different PRs anyway.  at this point, I would save just the files you changed, git checkout master and git pull, then make 2 different branches and manually put your changes in.

**Comment by rudri182** [comment]
May I close this PR and make a new one for the prerequisite section? I have already created new PR for CFU section. #807

**Comment by Kundan28** [comment]
> May I close this PR and make a new one for the prerequisite section? I have already created new PR for CFU section. #807

Yeah, go for it.

**Pull request closed without merging**